--- a/dist/index.js
+++ b/dist/index.js
@@ -243,12 +243,14 @@ function prepareTools({
     if (tool.type === "provider-defined") {
       toolWarnings.push({ type: "unsupported-tool", tool });
     } else {
+      const t = tool.function ?? tool;
       openaiCompatTools.push({
         type: "function",
         function: {
-          name: tool.name,
-          description: tool.description,
-          parameters: tool.inputSchema
+          name: t.name ?? t.id,
+          description: t.description,
+          parameters: t.parameters ?? t.inputSchema,
+          strict: t.strict,
         }
       });
     }
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -227,12 +227,14 @@ function prepareTools({
     if (tool.type === "provider-defined") {
       toolWarnings.push({ type: "unsupported-tool", tool });
     } else {
+      const t = tool.function ?? tool;
       openaiCompatTools.push({
         type: "function",
         function: {
-          name: tool.name,
-          description: tool.description,
-          parameters: tool.inputSchema
+          name: t.name ?? t.id,
+          description: t.description,
+          parameters: t.parameters ?? t.inputSchema,
+          strict: t.strict,
         }
       });
     }
