diff --git a/index.bun.js b/index.bun.js
index 853e369bf4a3945b3df5a8528f8700c35b24253d..db0115a2c15fd2fb1ede136fdf3d87b9143abcc8 100644
--- a/index.bun.js
+++ b/index.bun.js
@@ -11803,6 +11803,8 @@ class MarkdownRenderable extends Renderable {
       return;
     if (this._content !== value) {
       this._content = value;
+      if (!this._streaming)
+        this._parseState = null;
       this.updateBlocks();
       this.requestRender();
     }
diff --git a/index.node.js b/index.node.js
index 30f831a584c5b4ed5282aef739e9b6735cc3e1e0..613f5735d860939c3829fdfc498a43c578d957a8 100644
--- a/index.node.js
+++ b/index.node.js
@@ -11802,6 +11802,8 @@ class MarkdownRenderable extends Renderable {
       return;
     if (this._content !== value) {
       this._content = value;
+      if (!this._streaming)
+        this._parseState = null;
       this.updateBlocks();
       this.requestRender();
     }
